### PR TITLE
Resolve Development Branch Comments

### PR DESCRIPTION
**Pull request by Ibrahim-Jarada**
Relates #33

### CODE REVIEW

**Review by AymanSami98 — APPROVED** [review]

**Review by AymanSami98 — APPROVED** [review]

**Review by zaher-aa — APPROVED** [review]

**Review by Jood80 — APPROVED** [review]

File: src/utilities/viewlift/requests/viewliftListUserWatched.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ const getAllWatches = async (
           },
         });
 
-      let { counter } = listUserWatchedData;
+      let counter = listUserWatchedData?.counter || 0;
```
Comment (Jood80) [inline src/utilities/viewlift/requests/viewliftListUserWatched.ts:38]: What's the point of this optional chaining with `||`?

File: src/database/fakeData/index.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ if (NODE_ENV === 'development') {
     (async (): Promise<void> => {
       await buildFakeData();
       console.log('Fake data has been built');
+      sequelize.close();
     })();
   } catch (err) {
     console.log(err);
```
Comment (Jood80) [inline src/database/fakeData/index.ts:17]: `logger.error("some message of your", err)`

File: src/database/fakeData/index.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ if (NODE_ENV === 'development') {
     (async (): Promise<void> => {
       await buildFakeData();
       console.log('Fake data has been built');
+      sequelize.close();
```
Comment (Jood80) [inline src/database/fakeData/index.ts:14]: How about `try-catch-finally` block?
you `.close()` your Sequelize connection at the end regardless if your try worked or not.
[see](https://www.programiz.com/javascript/try-catch-finally)

**Review by Jood80 — CHANGES_REQUESTED** [review]

File: src/utilities/watchesPerContent.ts [hunk 4]
```diff
@@ -74,13 +75,26 @@ WPCOutputDTO => {
...
       (accumulatedWatchesDurationPerContent)[contentId] = {
         contentId,
         watchedSeconds: duration,
-        watches: [{ contentId, streamId, duration }],
+        watches: [{
+          contentId,
+          streamId,
+          duration,
+          createdBy: 1,
+          updatedBy: 1,
```
Comment (Jood80) [inline src/utilities/watchesPerContent.ts:95]: For upcoming PR updates, that role shouldn't be hardcoded like this, use `enum` and call its role name here.

**Review by Jood80 — CHANGES_REQUESTED** [review]

**Review by RandSohail — APPROVED** [review]

File: src/utilities/viewlift/requests/viewliftListUserWatched.ts (hunk 1 above)
Comment (zaher-aa) [reply src/utilities/viewlift/requests/viewliftListUserWatched.ts:38]: it appears to be a typo,
I fixed it.

File: src/database/fakeData/index.ts (hunk 3 above)
Comment (zaher-aa) [reply src/database/fakeData/index.ts:14]: Yes, good idea

**Review by Jood80 — APPROVED** [review]
Well DONE!!

**Pull request merged**